src/scenic/core/regions.py

```diff
@@ -281,883 +306,2816 @@ def __hash__(self):
+        # save on future calls.
+        padded_height = 100*max(1,centerZ)*height
+        bounded_footprint = self.boundFootprint(centerZ, padded_height)
+
+        self._bounded_cache = centerZ, padded_height, bounded_footprint
+
+        return bounded_footprint
+
+    def boundFootprint(self, centerZ, height):
+        """ Cap the footprint of the object to a given height, centered at a given z.
+
+        Args:
+            centerZ: The resulting mesh will be vertically centered at this height.
+            height: The resulting mesh will have this height.
+        """
+        # Fall back on progressively higher buffering and simplification to 
+        # get the mesh to be a valid volume
+        tol_sizes = [None, 0.00001, 0.0001, 0.001, 0.01, 0.1, 1]
+
+        for tol_size in tol_sizes:
+            # If needed, buffer the multipolygon
+            if tol_size is None:
+                poly_obj = self.polygon
+            else:
+                poly_obj = self.polygon.buffer(tol_size).simplify(tol_size)
+
+            # Extract a list of the polygon
+            if isinstance(poly_obj, shapely.geometry.Polygon):
+                polys = [poly_obj]
+            else:
+                polys = list(poly_obj.geoms)
+
+            # Extrude the polygon to the desired height
+            vf = [trimesh.creation.triangulate_polygon(p, engine='earcut') for p in polys]
+            v, f = trimesh.util.append_faces([i[0] for i in vf], [i[1] for i in vf])
+            polygon_mesh = trimesh.creation.extrude_triangulation(vertices=v, faces=f, height=height)
+
+            # Translate the polygon mesh to the desired height.
+            polygon_mesh.vertices[:,2] += centerZ - polygon_mesh.bounding_box.center_mass[2]
+
+            # Check if we have a valid volume
+            if polygon_mesh.is_volume:
+                if tol_size is not None and tol_size >= 0.01:
+                    warnings.warn(f"Computing bounded footprint of polygon resulted in invalid volume, "
+                                  f"but was able to remedy this by buffering/simplifying polygon by {tol_size}")
+                break
+
+        if not polygon_mesh.is_volume:
+            raise RuntimeError("Computing bounded footprint of polygon resulted in invalid volume")
+
+        return MeshVolumeRegion(polygon_mesh, centerMesh=False)
+
+    def buffer(self, amount):
+        buffered_polygon = self.polygon.buffer(amount)
+
+        if isinstance(buffered_polygon, shapely.geometry.Polygon):
+            buffered_polygon = MultiPolygon([buffered_polygon])
+
+        return PolygonalFootprintRegion(polygon=buffered_polygon, name=self.name)
+
+    @cached_property
+    def isConvex(self):
+        return self.polygons.equals(self.polygons.convex_hull)
+
+class PathRegion(Region):
+    """A region composed of multiple polylines in 3D space.
+
+    One of points or polylines should be provided. If both are provided,
+    points will be used.
```

Comment:
It would be better to raise a `ValueError` if both are provided, since it can't possibly be intended.